homeassistant/components/vicare/entity.py

```diff
@@ -27,7 +35,44 @@ def __init__(
         if hasattr(device, "id"):
             self._attr_unique_id += f"-{device.id}"
 
-        self._attr_device_info = DeviceInfo(
+        if isinstance(device, HeatingCircuit):
+            self._attr_device_info = self._get_info_for_component(
+                device_config, device, "Circuit", device.getName()
```

Comment:
`device.getName` is not async safe.

Reply:
Can a constructor be async or how could I solve this?

Reply:
Can a constructor be async or how could I solve this?

Reply:
No, the constructor can't await. Probably get the name before creating the entity and pass the name to the entity when creating it.